src/core/dfn-panel.runtime.js

```diff
@@ -0,0 +1,172 @@
+// @ts-check
+if (document.respecIsReady) {
+  document.respecIsReady.then(() => dfnPanel());
+} else {
+  dfnPanel();
+}
+
+function dfnPanel() {
+  /** @type {HTMLElement} */
+  let panel;
+  document.body.addEventListener("click", event => {
+    const el = /** @type {HTMLElement} */ (event.target);
+
+    const action = deriveAction(el);
+    switch (action) {
+      case "show": {
+        if (panel) panel.remove();
+        /** @type {HTMLElement} */
+        const dfn = el.closest("dfn, .index-term");
+        panel = createPanel(dfn);
+        displayPanel(dfn, panel, { x: event.clientX, y: event.clientY });
+        break;
+      }
+      case "dock": {
+        panel.style.left = null;
+        panel.style.top = null;
+        panel.classList.add("docked");
+        break;
+      }
+      case "hide": {
+        panel.remove();
+        break;
+      }
+    }
+  });
+}
+
+/** @param {HTMLElement} clickTarget */
+function deriveAction(clickTarget) {
+  const hitALink = !!clickTarget.closest("a");
+  if (clickTarget.closest("dfn, .index-term")) {
+    return hitALink ? null : "show";
+  }
+  if (clickTarget.closest("#dfn-panel")) {
+    if (hitALink) {
+      const clickedSelfLink = clickTarget.classList.contains("self-link");
+      return clickedSelfLink ? "hide" : "dock";
+    }
+    const panel = clickTarget.closest("#dfn-panel");
+    return panel.classList.contains("docked") ? "hide" : null;
+  }
+  if (document.getElementById("dfn-panel")) {
+    return "hide";
+  }
+  return null;
+}
+
+/** @param {HTMLElement} dfn */
+function createPanel(dfn) {
+  const { id } = dfn;
+  const href = dfn.dataset.href || `#${id}`;
+  /** @type {NodeListOf<HTMLAnchorElement>} */
+  const links = document.querySelectorAll(`a[href="${href}"]:not(.index-term)`);
+
+  const tempNode = document.createElement("div");
+  tempNode.innerHTML = `
+    <aside class="dfn-panel" id="dfn-panel">
+      <span class="caret"></span>
+      <b><a class="self-link" href="${href}">Permalink</a></b>
```

Comment:
Nit: do we really need `<b>` ?

Reply:
Looked nice IMO. Also, we're using it as `display: block` :sweat_smile: